shared-bindings/sdcardio/SDCard.c

```diff
@@ -0,0 +1,180 @@
+/*
+ * This file is part of the Micro Python project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Jeff Epler for Adafruit Industries
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/obj.h"
+#include "py/objproperty.h"
+#include "py/runtime.h"
+#include "py/objarray.h"
+
+#include "shared-bindings/sdcardio/SDCard.h"
+#include "shared-module/sdcardio/SDCard.h"
+#include "common-hal/busio/SPI.h"
+#include "shared-bindings/busio/SPI.h"
+#include "shared-bindings/microcontroller/Pin.h"
+#include "supervisor/flash.h"
+
+//| .. currentmodule:: sdcardio
+//|
+//| :class:`SDCard` --  SD Card Block Interface
+//| ===========================================
+//|
+//| Controls an SD card over SPI.  This built-in module has higher read
+//| performance than the library adafruit_sdcard, but it is only compatible with
+//| `busio.SPI`, only `bitbangio.SPI`.  Usually an SDCard object is used
+//| with `storage.VfsFat` to allow file I/O to an SD card.
+//|
+//| .. class:: SDCard(spi, cs, baudrate)
+//|
+//|   :param busio.SPI spi: The SPI bus
+//|   :param microcontroller.Pin cs: The chip select connected to the card
+//|   :param int baudrate: The SPI data rate to use after card setup
+//|
+//|   Example usage:
+//|
+//|   .. code-block:: python
+//|
+//|       import os
+//|
+//|       import board
+//|       import sdcardio
+//|       import storage
+//|
+//|       sd = adafruit_sdcard.SDCard(board.SPI(), board.SD_CS)
```

Comment:
Is this example correct? adafruit_sdcard is not imported.. am I just missing somthing?

Reply:
Better?